runtime/port/sysvipc/j9shmem.c

```diff
@@ -1242,30 +1243,90 @@ getControlFilePath(struct J9PortLibrary* portLibrary, const char* cacheDirName,
 }
 
 intptr_t
-j9shmem_getDir(struct J9PortLibrary* portLibrary, const char* ctrlDirName, BOOLEAN appendBaseDir, char* buffer, uintptr_t bufLength)
+j9shmem_getDir(struct J9PortLibrary* portLibrary, const char* ctrlDirName, uint32_t flags, char* buffer, uintptr_t bufLength)
 {
 	OMRPORT_ACCESS_FROM_J9PORT(portLibrary);
-	const char* rootDir = J9SH_DEFAULT_CTRL_ROOT;
+	const char* rootDir = NULL;
+	char homeDirBuf[J9SH_MAXPATH];
+	const char* homeDir = NULL;
+	BOOLEAN appendBaseDir = J9_ARE_ALL_BITS_SET(flags, J9SHMEM_GETDIR_APPEND_BASEDIR);
 
 	Trc_PRT_j9shmem_getDir_Entry();
 
+	memset(homeDirBuf, 0, J9SH_MAXPATH);
+
 	if (ctrlDirName != NULL) {
 		rootDir = ctrlDirName;
-	}
-
-	if (appendBaseDir) {
-		if (omrstr_printf(buffer, bufLength, "%s/%s", rootDir, J9SH_BASEDIR) == bufLength - 1) {
-			Trc_PRT_j9shmem_getDir_ExitFailedOverflow();
-			return -1;
+	} else {
+		if (J9_ARE_ALL_BITS_SET(flags, J9SHMEM_GETDIR_USE_USERHOME)) {
+			Assert_PRT_true(TRUE == appendBaseDir);
+			uintptr_t baseDirLen = strlen(J9SH_BASEDIR);
+			/* From https://linux.die.net/man/3/getpwuid:
+			 * "An application that wants to determine its user's home directory should inspect the value of HOME (rather than the value getpwuid(getuid())->pw_dir)
```

Comment:
Please don't copy the text from the Linux man pages. You can paraphrase.

Reply:
Fixed